File: NET_4.5.2/C1.WPF.FlexGrid/CS/ExcelBook/C1.WPF.FlexGridBook/ColorScheme.cs
using System;
using System.Windows;
using System.Windows.Media;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using C1.WPF.FlexGrid;

namespace C1.WPF.FlexGridBook
{
    /// <summary>
    /// Specifies the color scheme used to display the control.
    /// </summary>
    public enum ColorScheme
    {
        /// <summary>
        /// Office 2007 blue color scheme.
        /// </summary>
        Blue,
        /// <summary>
        /// Office 2007 silver color scheme.
        /// </summary>
        Silver,
        /// <summary>
        /// Office 2007 black color scheme.
        /// </summary>
        Black
    };

    /// <summary>
    /// Class that defines and applies color schemes to C1FlexGrid controls.
    /// </summary>
    internal static class ColorSchemeManager
    {
        static Brush _brWhite = new SolidColorBrush(Colors.White);

        public static void ApplyColorScheme(C1FlexGrid grid, ColorScheme cs)
        {
            var c = PaletteProvider.GetPalette(cs);
            if (c != null)
            {
                grid.CursorBackground =
                grid.EditorBackground = null;
                grid.RowBackground =
                grid.AlternatingRowBackground =
                grid.GroupRowBackground = _brWhite;
                grid.TopLeftCellBackground = new SolidColorBrush(c[0]);
                grid.RowHeaderBackground = new SolidColorBrush(c[1]);
                grid.RowHeaderSelectedBackground = new SolidColorBrush(c[2]);
                grid.ColumnHeaderBackground = CreateGradientBrush(c[3], c[4]);
                grid.ColumnHeaderSelectedBackground = CreateGradientBrush(c[5], c[6]);
                grid.GridLinesBrush = new SolidColorBrush(c[7]); ;
                grid.HeaderGridLinesBrush = new SolidColorBrush(c[8]);
                grid.SelectionBackground = new SolidColorBrush(c[9]);
            }
        }
        static LinearGradientBrush CreateGradientBrush(Color top, Color bot)
        {
            var lgb = new LinearGradientBrush();
            var gsc = lgb.GradientStops;
            gsc.Add(new GradientStop() { Color = top, Offset = 0 });
            gsc.Add(new GradientStop() { Color = bot, Offset = 1 });
            lgb.EndPoint = new Point(0, 1);
            return lgb;
        }
        public static class PaletteProvider
        {
            public static Color[] GetPalette(ColorScheme scheme)
            {
                switch (scheme)
                {
                    case ColorScheme.Blue:
                        return new Color[]
                        {
                            Color.FromArgb(0xff, 0xa9, 0xc4, 0xe9), // brTopLeft
                            Color.FromArgb(0xff, 0xe4, 0xec, 0xf7), // brRowHdr
                            Color.FromArgb(0xff, 0xff, 0xd5, 0x8d), // brRowHdrSel 
                            Color.FromArgb(0xff, 0xf6, 0xfa, 0xfb), // brColHdr (top)
                            Color.FromArgb(0xff, 0xd5, 0xdd, 0xea), // brColHdr (bot)
                            Color.FromArgb(0xff, 0xf8, 0xd7, 0x9b), // brColHdrSel (top)
                            Color.FromArgb(0xff, 0xf1, 0xc2, 0x63), // brColHdrSel (bot)
                            Color.FromArgb(0xff, 0xd0, 0xd7, 0xe5), // brLines
                            Color.FromArgb(0xff, 0x9e, 0xb6, 0xce), // brHdrLines
                            Color.FromArgb(0xff, 0xea, 0xec, 0xf5)  // brSelection 
                        };
                    case ColorScheme.Silver:
                        return new Color[] 
                        {
                            Color.FromArgb(0xff, 0xc6, 0xc6, 0xc6), // brTopLeft
                            Color.FromArgb(0xff, 0xe7, 0xe7, 0xe7), // brRowHdr
                            Color.FromArgb(0xff, 0xf5, 0xc7, 0x95), // brRowHdrSel 
                            Color.FromArgb(0xff, 0xf1, 0xf3, 0xf3), // brColHdr (top)
                            Color.FromArgb(0xff, 0xc8, 0xc9, 0xca), // brColHdr (bot)
                            Color.FromArgb(0xff, 0xff, 0xc9, 0x96), // brColHdrSel (top)
                            Color.FromArgb(0xff, 0xff, 0x9b, 0x68), // brColHdrSel (bot)
                            Color.FromArgb(0xff, 0xd0, 0xd7, 0xe5), // brLines
                            Color.FromArgb(0xff, 0x90, 0x91, 0x92), // brHdrLines
                            Color.FromArgb(0xff, 0xea, 0xec, 0xf5)  // brSelection 
                        };
                    case ColorScheme.Black:
                        return new Color[]
                        {
                            Color.FromArgb(0xff, 0xc9, 0xc9, 0xc9), // brTopLeft
                            Color.FromArgb(0xff, 0xed, 0xed, 0xed), // brRowHdr
                            Color.FromArgb(0xff, 0xff, 0xd5, 0x8d), // brRowHdrSel 
                            Color.FromArgb(0xff, 0xf6, 0xf6, 0xf6), // brColHdr (top)
                            Color.FromArgb(0xff, 0xde, 0xde, 0xde), // brColHdr (bot)
                            Color.FromArgb(0xff, 0xf8, 0xd7, 0x9b), // brColHdrSel (top)
                            Color.FromArgb(0xff, 0xf1, 0xc1, 0x5f), // brColHdrSel (bot)
                            Color.FromArgb(0xff, 0xd0, 0xd7, 0xe5), // brLines
                            Color.FromArgb(0xff, 0xb6, 0xb6, 0xb6), // brHdrLines
                            Color.FromArgb(0xff, 0xea, 0xec, 0xf5)  // brSelection 
                        };
                }

                // invalid color scheme
                return null;
            }
        }
    }
}

File: NET_4.5.2/C1.WPF.FlexGrid/VB/ExcelGrid/ExcelGrid/FlexExcel/ColorScheme.vb
Imports System.Text
Imports C1.WPF.FlexGrid

Namespace ExcelGrid
	''' <summary>
	''' Specifies the color scheme used to display the control.
	''' </summary>
	Public Enum ColorScheme
		''' <summary>
		''' Office 2007 blue color sheme.
		''' </summary>
		Blue
		''' <summary>
		''' Office 2007 silver color sheme.
		''' </summary>
		Silver
		''' <summary>
		''' Office 2007 black color sheme.
		''' </summary>
		Black
	End Enum

	''' <summary>
	''' Class that defines and applies color schemes to C1FlexGrid controls.
	''' </summary>
	Public NotInheritable Class ColorSchemeManager
		Private Sub New()
		End Sub
		Public Shared Sub ApplyColorScheme(ByVal grid As C1FlexGrid, ByVal cs As ColorScheme)
			Dim c = PaletteProvider.GetPalette(cs)
			If c IsNot Nothing Then
				grid.EditorBackground = New SolidColorBrush(Colors.Transparent)
				grid.CursorBackground = grid.EditorBackground
				grid.AlternatingRowBackground = grid.CursorBackground
				grid.RowBackground = grid.AlternatingRowBackground
				grid.TopLeftCellBackground = New SolidColorBrush(c(0))
				grid.RowHeaderBackground = New SolidColorBrush(c(1))
				grid.RowHeaderSelectedBackground = New SolidColorBrush(c(2))
				grid.ColumnHeaderBackground = CreateGradientBrush(c(3), c(4))
				grid.ColumnHeaderSelectedBackground = CreateGradientBrush(c(5), c(6))
				grid.GridLinesBrush = New SolidColorBrush(c(7))

				grid.HeaderGridLinesBrush = New SolidColorBrush(c(8))
				grid.SelectionBackground = New SolidColorBrush(c(9))
			End If
		End Sub
		Private Shared Function CreateGradientBrush(ByVal top As Color, ByVal bot As Color) As LinearGradientBrush
			Dim lgb = New LinearGradientBrush()
			Dim gsc = lgb.GradientStops
			gsc.Add(New GradientStop() With {.Color = top, .Offset = 0})
			gsc.Add(New GradientStop() With {.Color = bot, .Offset = 1})
			lgb.EndPoint = New Point(0, 1)
			Return lgb
		End Function
		Public NotInheritable Class PaletteProvider
			Private Sub New()
			End Sub
			Public Shared Function GetPalette(ByVal scheme As ColorScheme) As Color()
				Select Case scheme
					Case ColorScheme.Blue
						Return New Color() { Color.FromArgb(&Hff, &Ha9, &Hc4, &He9), Color.FromArgb(&Hff, &He4, &Hec, &Hf7), Color.FromArgb(&Hff, &Hff, &Hd5, &H8d), Color.FromArgb(&Hff, &Hf6, &Hfa, &Hfb), Color.FromArgb(&Hff, &Hd5, &Hdd, &Hea), Color.FromArgb(&Hff, &Hf8, &Hd7, &H9b), Color.FromArgb(&Hff, &Hf1, &Hc2, &H63), Color.FromArgb(&Hff, &Hd0, &Hd7, &He5), Color.FromArgb(&Hff, &H9e, &Hb6, &Hce), Color.FromArgb(&Hff, &Hea, &Hec, &Hf5) }
					Case ColorScheme.Silver
						Return New Color() { Color.FromArgb(&Hff, &Hc6, &Hc6, &Hc6), Color.FromArgb(&Hff, &He7, &He7, &He7), Color.FromArgb(&Hff, &Hf5, &Hc7, &H95), Color.FromArgb(&Hff, &Hf1, &Hf3, &Hf3), Color.FromArgb(&Hff, &Hc8, &Hc9, &Hca), Color.FromArgb(&Hff, &Hff, &Hc9, &H96), Color.FromArgb(&Hff, &Hff, &H9b, &H68), Color.FromArgb(&Hff, &Hd0, &Hd7, &He5), Color.FromArgb(&Hff, &H90, &H91, &H92), Color.FromArgb(&Hff, &Hea, &Hec, &Hf5) }
					Case ColorScheme.Black
						Return New Color() { Color.FromArgb(&Hff, &Hc9, &Hc9, &Hc9), Color.FromArgb(&Hff, &Hed, &Hed, &Hed), Color.FromArgb(&Hff, &Hff, &Hd5, &H8d), Color.FromArgb(&Hff, &Hf6, &Hf6, &Hf6), Color.FromArgb(&Hff, &Hde, &Hde, &Hde), Color.FromArgb(&Hff, &Hf8, &Hd7, &H9b), Color.FromArgb(&Hff, &Hf1, &Hc1, &H5f), Color.FromArgb(&Hff, &Hd0, &Hd7, &He5), Color.FromArgb(&Hff, &Hb6, &Hb6, &Hb6), Color.FromArgb(&Hff, &Hea, &Hec, &Hf5) }
				End Select

				' invalid color scheme
				Return Nothing
			End Function
		End Class
	End Class
End Namespace
